frontend/src/Components/RoadDetails/ImageGallery.tsx

```diff
@@ -70,18 +70,17 @@ const images: Image[] = [
 
 const ImageGallery: React.FC = () => {
   const [scrollPosition, setScrollPosition] = useState(0);
+  const [centerImages, setCenterImages] = useState(false);
+  const galleryRef = useRef<HTMLDivElement>(null);
 
   const openImageInNewTab = (imageUrl: string) => {
     window.open(imageUrl, '_blank');
   };
 
   const handleScroll = (scrollOffset: number) => {
     const newScrollPosition = scrollPosition + scrollOffset;
+    const maxScrollLimit = (images.length - 11.4) * 120;
```

Comment:
I think the issue of being able to scroll after the images is because of the `11.4`. From my understanding, this can be interpreted as the number of images on the screen of the user... But this number depends on the width of the screen of the user.

Also, I'm not sure that the width (`120`px) will be the same for the real image nor that they will all have the same width...

I think it might be easier to do something with a variable storing the index of the current image at the left of the screen (the current first image). Then the width to translate would be the sum of the width of the images (to get with the html element) and the margins. And the max index would be the first one verifying `screen.width - sum from index to the last image of (image width + margin) >= 0` (meaning the first index where the width occupied by the images from index to the end is smaller than the width of the screen)

Reply:
To be clear, I'm not saying that it can't be done with the current mechanism, I'm saying that I think it would be easier to do it like that and that would probably be the way I would do it.